RetailCoder.VBE/Settings/HotkeySettings.cs

```diff
@@ -61,14 +42,23 @@ public HotkeySetting[] Settings
                 }
 
                 //Merge any hotkeys that weren't found in the input.
-                foreach (var setting in Defaults.Where(setting => _settings.FirstOrDefault(s => s.Name.Equals(setting.Name)) == null))
+                foreach (var setting in defaults.Where(setting => _settings.FirstOrDefault(s => s.Name.Equals(setting.Name)) == null))
                 {
                     setting.IsEnabled &= !IsDuplicate(setting);
                     _settings.Add(setting);
                 }
             }
         }
 
+        public HotkeySettings()
+        {
+        }
+
+        public HotkeySettings(IEnumerable<HotkeySetting> defaultSettings)
+        {
+            _defaultSettings = defaultSettings;
+        }
```

Comment:
initially I commented on this constructor that this was not a concern of `HotkeySettings`, but I now noticed that it exactly is. What the constructor does though is something a bit different. It makes grabbing the default settings a concern of Castle Windsor, which is not really what our IoC container should be doing.

Also this current configuration allows just creating HotkeySettings without any defaultSettings. When attempting to run `Settings`, the default constructor is what necessitates the null-coalescing operator. Solution: either drop the default ctor or initialize `_defaultSettings` in the zero-argument ctor

Reply:
Removing the default constructor is not an option, because it is required for deserialization. I don't think initializing `_defaultSettings` will allow me to remove `??`. The operator is there in case `_settings` are null. There is a bug here anyway, though. After calling the default constructor both fields are null, so the getter will throw an exception.